src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java

```diff
@@ -39,34 +42,53 @@ public class AddAppointmentCommand extends Command {
     private final String description;
     private final LocalDateTime dateTime;
 
+    private final Index doctorIndex;
+
     /**
      * Creates an AddCommand to add the specified {@code Person}
      */
-    public AddAppointmentCommand(Index targetIndex, String description, LocalDateTime dateTime) {
+    public AddAppointmentCommand(Index targetIndex, Index doctorIndex, String description, LocalDateTime dateTime) {
         this.targetIndex = targetIndex;
         this.description = description;
         this.dateTime = dateTime;
+        this.doctorIndex = doctorIndex;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Person> lastShownList = model.getFilteredPersonList();
+        List<Doctor> lastDoctorList = model.getFilteredDoctorList();
+
+        if (lastDoctorList.size() == 0) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
```

Comment:
can we change the error message displayed? currently if no doctors are available, error message thrown is "invalid patient Index"

"Invalid doctor index" would be a more suitable error message